### ISSUE TITLE
Add option to customize theme

### ISSUE THREAD

**Issue by rafaelrinaldi** [rafaelrinaldi/pure#2]
Since we already use variables everywhere, it should be as simple as changing them from local to the global scope.

**Comment by rafaelrinaldi** [comment]
@oranja @edouard-lopez what's the best way to achieve this?

**Comment by oranja** [comment]
For bare fish (not using any framework), I believe the most standard way is to rearrange the variable assignments to serve as defaults in case the variable is not yet set.
For example:
```fish
# Symbols
set -q symbol_prompt; or set -g symbol_prompt "❯"
set -q symbol_git_down_arrow; or set -g symbol_git_down_arrow "⇣"
set -q symbol_git_up_arrow; or set -g symbol_git_up_arrow "⇡"
set -q symbol_git_dirty; or set -g symbol_git_dirty "*"
set -q symbol_horizontal_bar; or set -g symbol_horizontal_bar "—"
```
This way, any of these variables can be defined in **config.fish** without being overwritten by pure's fish_prompt.fish.

If it gets too tedious, you can define a helper function:
```fish
function set_default -S -a var value
    set -q $var; or set -g $var $value
end
```
Then the previous block becomes:
```fish
# Symbols
set_default symbol_prompt "❯"
set_default symbol_git_down_arrow "⇣"
set_default symbol_git_up_arrow "⇡"
set_default symbol_git_dirty "*"
set_default symbol_horizontal_bar "—"
```
In the price of indirection you get a more compact block, and it also helps to avoid bugs from typos...

**Comment by rafaelrinaldi** [comment]
@oranja that looks pretty decent and straightforward. Will give it a try! Thanks man.

**Comment by bucaran** [comment]
Agree with @oranja, but please always prefix any global variables with the name of your plugin/theme/prompt/snippet/etc so as to not conflict (hardly, but still) with other global variables of the same name.

**Comment by rafaelrinaldi** [comment]
Great, I will keep that in mind. Thanks.

--  
Rafael Rinaldi
http://rinaldi.io

**Comment by edouard-lopez** [comment]
Thanks for your knowledge!

@oranja good proposal
@bucaran nice improvement! Love future-proof argument!
don't we need `-g` to set globally?

**Comment by oranja** [comment]
What @bucaran is suggesting is to have two variables for each customization options.
The first is local and is the one actually used in the theme code.
The second can have any scope that affects the prompt - universal, global, or local to an outer function (i.e. a test function).
To clarify, here's @bucaran's suggestion again, with annotations. (hoping I won't get sued for copyright infringement here).
```fish
set -l symbol_prompt ❯    # This is the local variable that will be used in the prompt's code.

if set -q pure_symbol_prompt    # Here we check for any previous definition of `pure_symbol_prompt`. Not necessarily global.
    set symbol_prompt $pure_symbol_prompt    # We override the local variable with the external value, if it's present.
end
```

**Comment by bucaran** [comment]
One more thing that I don't know whether it supports my argument or not, is that fish does this too, so it's a way to stay consistent across the universe.

https://github.com/fish-shell/fish-shell/blob/master/share/functions/__fish_config_interactive.fish#L18

> don't we need -g to set globally?

Depends. I was assuming this variable will only be used in the file where it was declared.

**Comment by oranja** [comment]
In the same file fish also has several of these `set -q var; or set var`
one liners, rendering your choice argument weaker.
Having said that, both are fine, but personally, I think that if you have
many of these in a row, it's better to use the shorter form. It's an
equally acceptable fish idiom.

**Comment by Shougo** [comment]
I think this issue should be closed by #30.

**Comment by rafaelrinaldi** [comment]
@Shougo That's true. Thanks!
